### PR TITLE
Added precision to bounding box with custom round function

### PR DESCRIPTION
**Pull request by riccardogiorato**
Added precision to bounding box with custom round function.

This enables to get accurate sizes for objects.

Current users won't notice the change cause with default param the module API will act the same.

### CODE REVIEW

**Comment by risq** [comment]
Thank you for your PR !!

Could you please add a few unit tests to test the precision option with the example model ?

Thanks.

File: src/glb2-bounding-box.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import gltf2BoundingBox from './gltf2-bounding-box';
 
 const glb2BoundingBox = {
 
-  computeBoundings(glb) {
+  computeBoundings(glb,precision=0) {
```
Comment (risq) [inline src/glb2-bounding-box.js]: Could you use a "parameters" object ? We may add more options in the future.

Something like

```js
  computeBoundings(glb, { precision = 0 } = {}) {
```

File: src/gltf1-bounding-box.js [hunk 2]
```diff
@@ -2,9 +2,11 @@ import { Matrix } from 'matrixmath';
 import { flattenDeep, includes } from 'lodash';
 import { loadPositions } from './gltf-reader';
 
+import precise from './precise';
+
 const gltf1BoundingBox = {
 
-  computeBoundings(gltf) {
+  computeBoundings(gltf,precision=0) {
```
Comment (risq) [inline src/gltf1-bounding-box.js]: Same here.

**Review by risq — CHANGES_REQUESTED** [review]
By the way you don't need to include the dist files into the PR. They will be built in the following release !

**Comment by riccardogiorato** [comment]
I added dist build cause I wanted to use them as soon as possible on my projects. About tests, I tried fixing them but it kept failing with NaN results. Could it be caused by spy?

**Comment by risq** [comment]
Indeed, I have investigated a little and I found out that the main function was passing a buffer object to gltf1 & glb functions as a second parameter, which was used as the precision in the `computeBoundings` methods.

I just pushed a fix on master, tell me if you still have issues with your tests.

**Comment by riccardogiorato** [comment]
Your commit is still failing the build: "The Travis CI build failed"

**Comment by risq** [comment]
OK, it is fixed now, it was an issue between travis and some of the dependencies of gulp.

**Comment by riccardogiorato** [comment]
Can you merge this pull request? It might be really usefuel to have the variable precision for 3d models

**Comment by aronfiechter** [comment]
Will this PR be merged? I'm not sure what the issue is with the failing build, but I agree with @Giorat that a precision parameter would be really helpful.

**Comment by aronfiechter** [comment]
I forked the fork and corrected some issues, should I create a pull request with (my changes) to this repo or to @Giorat's fork?

**Comment by riccardogiorato** [comment]
To mine so then I can merge into this one! 😄

**Comment by aronfiechter** [comment]
@Giorat done

**Comment by riccardogiorato** [comment]
> @Giorat done

Still present the issue

**Comment by aronfiechter** [comment]
> Still present the issue

Yes, there are still merge conflicts, but we cannot do anything about those.
But the tests are now all passing

@risq what are the conflicts in `index.js` about?

**Comment by aronfiechter** [comment]
@risq the build fails likely because of a problem with `gulp` and `node 11`: https://github.com/gulpjs/gulp/issues/2246

**Pull request merged**